python/ray/test_utils.py

```diff
@@ -370,6 +370,13 @@ def put_object(obj, use_ray_put):
         return _put.remote(obj)
 
 
+def put_object_unpinned(obj):
+    value = ray.worker.global_worker.get_serialization_context().serialize(obj)
+    return ray.ObjectRef(
+        ray.worker.global_worker.core_worker.put_serialized_object(
+            value, pin_object=False))
```

Comment:
New testing utility for creating an unpinned object. Given that creating an unpinned primary (and only) copy of an object doesn't fit with the [object memory management model](https://docs.google.com/document/d/1lAy0Owi-vPz2jEqBSaHNQcy2IBSDEHyXNOQZlGuj93c/preview#heading=h.61xcnifjkb6v), I thought that it'd be better if creating such an unpinned object was exposed in a testing utility rather than through the externally facing API.